### ISSUE TITLE
go1.10.windows-{386,amd64} zip distributions contain "go.exe~" and "dist.exe~"

### ISSUE THREAD

**Issue by DHowett** [golang/go#23894]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
1.10

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?
386, amd64

### What did you do?

Download the go1.10.windows-amd64.zip archive. Extract it or otherwise inspect its contents.
It seems to contain `go/bin/go.exe~`

```
% unzip -l go1.10.windows-amd64.zip | grep "~$"
 11180544  2018-02-16 09:14   go/bin/go.exe~
  3551744  2018-02-16 09:14   go/pkg/tool/windows_amd64/dist.exe~
```

**Comment by dgryski** [comment]
@andybons

**Comment by larrylau** [comment]
what go.exe~ and  dist.exe~ can do?
go1.10.1 same.

**Comment by andybons** [comment]
This is a matter of simply not cleaning up everything when building the release via our automated process. This will be fixed.

**Comment by vbmade2000** [comment]
@andybons Is this issue open for work ?

**Comment by andybons** [comment]
@vbmade2000 meaning is it available for you to work on it? Sure.

**Comment by vbmade2000** [comment]
@andybons Cool. Thanks.

**Comment by andybons** [comment]
@alexbrainman any idea what’s going on?

**Comment by vbmade2000** [comment]
@andybons Nope. Didn't get much time to investigate it. I'll check today.

**Comment by alexbrainman** [comment]
> @alexbrainman any idea what’s going on?

go.exe~ file is created when go.exe rebuilds itself.

When Windows process is running, its executable cannot be deleted. So how would go.exe replace its executable when it needs to rebuild itself? As we discovered long time ago, you cannot delete executable, but it is OK to rename executable. So that is what we do - when go.exe try to replace itself and fails, it renames go.exe into go.exe~, and then it can write new go.exe. That leaves go.exe~ behind, because cmd/go process still cannot remove go.exe~. go.exe~ can be removed later once its process exits, but I suspect we never bother to do that.

It would be the same story for dist.exe~.

You can search for `On Windows, remove lingering ~ file from last attempt.` in your GOROOT for details.

I suspect (I did not actually check) you should just modify release.go to have postBuildCleanFiles include both go.exe~ and dist.exe~ files.

I hope it helps.

Alex

**Comment by vbmade2000** [comment]
I was able to reproduce the issue. There are files called `go.exe~` and `dist.exe~` in [go1.10.2.windows-amd64.zip](https://dl.google.com/go/go1.10.2.windows-amd64.zip) file. 

I checked SHA256 value and found it same as given on download site.
The `go.exe~` is a go executable showing version 1.10.2. File size is less than original go.exe file.
I tried to check msi file by extracting it but didn't get sensible output to come to some conclusion.

**Comment by vbmade2000** [comment]
@alexbrainman Makes sense. I want to test your suggestion. Can you tell me how to do that on local system ?

**Comment by andybons** [comment]
@alexbrainman that helps immensely. Thank you :)

**Comment by andybons** [comment]
@vbmade2000 yep. go for it.
